### PR TITLE
trigger one prod job per run

### PR DESCRIPTION
**Pull request by andreer**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/EndpointCertificateMaintainer.java [hunk 1]
```diff
@@ -93,17 +94,18 @@ private void updateRefreshedCertificates() {
...
         var now = clock.instant();
+        var jobsTriggered = new AtomicInteger(0);
         curator.readAllEndpointCertificateMetadata().forEach((applicationId, endpointCertificateMetadata) ->
                 endpointCertificateMetadata.lastRefreshed().ifPresent(lastRefreshTime -> {
                     Instant refreshTime = Instant.ofEpochSecond(lastRefreshTime);
                     if (now.isAfter(refreshTime.plus(4, ChronoUnit.DAYS))) {
                         controller().applications().getInstance(applicationId)
                                 .ifPresent(instance -> instance.productionDeployments().forEach((zone, deployment) -> {
-                                    if (deployment.at().isBefore(refreshTime)) {
+                                    if (deployment.at().isBefore(refreshTime) && jobsTriggered.compareAndSet(0, 1)) {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/EndpointCertificateMaintainer.java:108]: Perhaps randomise the order, lest a single, failing deployment blocks all others?

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/EndpointCertificateMaintainer.java:108]: Or better still, pick the eligible deployment with the oldest `lastTriggered` timestamp?

Comment (andreer) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/EndpointCertificateMaintainer.java]: That makes sense

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/EndpointCertificateMaintainer.java [hunk 2]
```diff
@@ -93,27 +95,35 @@ private void updateRefreshedCertificates() {
         }));
     }
 
+    record EligibleJob(Deployment deployment, ApplicationId applicationId, JobType job) {}
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/EndpointCertificateMaintainer.java:98]: Wohoo!

**Review by jonmv — APPROVED** [review]
Nice :)

**Pull request merged**